src/haywire/http_server.c

```diff
@@ -275,6 +293,86 @@ void http_stream_on_read_http_parser(uv_stream_t* tcp, ssize_t nread, const uv_b
     free(buf->base);
 }
 
+void http_stream_on_read_pico(uv_stream_t* tcp, ssize_t nread, const uv_buf_t* buf)
+{
+    http_connection* connection = (http_connection*)tcp->data;
+    
+    // TODO: nread == 0 should mean the socket closed. We should handle this properly.
```

Comment:
Can it not also mean that we're waiting for the client to send more data, and there's none available since we're running in non-blocking mode?